From 8d0506f82d1b5cf46f03140058955e00ff8e3e11 Mon Sep 17 00:00:00 2001
From: wei.wang1 <wei.wang1@amlogic.com>
Date: Thu, 03 Mar 2022 16:28:10 +0800
Subject: [PATCH] Audio: In YouTube Audio is Not there and playback stuck [1/1]

PD#SWPL-68939

Problem:
clock is wrong, cause no sound.

Solution:
correct clock.

Verify:
ap223 rdk.

Change-Id: Id12e69c07435aae2acf068bdbe2634992738403b
Signed-off-by: wei.wang1 <wei.wang1@amlogic.com>
---

diff --git a/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts b/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts
index 464d1f2..6f09ea5 100644
--- a/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts
+++ b/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts
@@ -1020,8 +1020,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
@@ -1075,8 +1075,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
diff --git a/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts b/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts
index c260c13..7b6e6f8 100644
--- a/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts
+++ b/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts
@@ -1020,8 +1020,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
@@ -1075,8 +1075,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
diff --git a/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts b/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts
index 3cf5f85..304d06d 100644
--- a/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts
+++ b/arch/arm/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts
@@ -1012,8 +1012,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
@@ -1067,8 +1067,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
diff --git a/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts b/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts
index 04692cf..321732c 100644
--- a/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts
+++ b/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm.dts
@@ -1022,8 +1022,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
@@ -1077,8 +1077,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
diff --git a/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts b/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts
index e5a4cc5..610ffe3 100644
--- a/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts
+++ b/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_drm_s.dts
@@ -1022,8 +1022,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
@@ -1077,8 +1077,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
diff --git a/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts b/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts
index 455dae3..b0c9581 100644
--- a/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts
+++ b/arch/arm64/boot/dts/amlogic/s4d_s905y4_ap223_linux.dts
@@ -1014,8 +1014,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
@@ -1069,8 +1069,8 @@
 			&clkc CLKID_MPLL3
 			&clkaudio CLKID_AUDIO_PDMIN0
 			&clkaudio CLKID_AUDIO_PDMIN1
-			&clkc CLKID_MPLL3
-			&clkaudio CLKID_AUDIO_MCLK_B>;
+			&clkc CLKID_MPLL0
+			&clkaudio CLKID_AUDIO_MCLK_A>;
 		clock-names = "pdm_gate",
 			"pdm_sysclk_srcpll",
 			"pdm_dclk_srcpll",
